src/internet_identity/tests/integration/activity_stats/authn_methods.rs

```diff
@@ -0,0 +1,276 @@
+use crate::v2_api::authn_method_test_helpers::{
+    create_identity_with_authn_method, test_authn_method,
+};
+use canister_tests::api::internet_identity::api_v2;
+use canister_tests::framework::{
+    assert_labelled_metric, assert_metric, env, get_metrics, install_ii_canister,
+    upgrade_ii_canister, II_WASM,
+};
+use ic_test_state_machine_client::CallError;
+use internet_identity_interface::internet_identity::types::{
+    AuthnMethod, AuthnMethodData, MetadataEntry, Purpose, WebAuthn,
+};
+use serde_bytes::ByteBuf;
+use std::collections::HashMap;
+use std::time::Duration;
+
+const DAY_SECONDS: u64 = 24 * 60 * 60;
+const MONTH_SECONDS: u64 = 30 * DAY_SECONDS;
+
+const AUTHN_METHOD_TYPES: [&str; 5] = [
+    "other",
+    "webauthn_auth",
+    "webauthn_recovery",
+    "recovery_phrase",
+    "browser_storage_key",
+];
+
+/// Tests that daily active authn_methods are counted correctly.
+#[test]
+fn should_report_daily_active_authn_methods() -> Result<(), CallError> {
+    let env = env();
+    let canister_id = install_ii_canister(&env, II_WASM.clone());
+
+    // ensure stats are initially absent
+    assert!(
+        !get_metrics(&env, canister_id).contains("internet_identity_daily_active_authn_methods")
+    );
+
+    for (authn_type, authn_method) in authn_methods_all_types() {
+        // advance time to ensure a new collection period
+        env.advance_time(Duration::from_secs(DAY_SECONDS));
+
+        create_identity_with_authn_method(&env, canister_id, &authn_method);
+        let identity_nr = create_identity_with_authn_method(&env, canister_id, &authn_method);
```

Comment:
why calling `create_identity_with_authn_method` twice and ignoring the first identity?  (maybe add a comment)

Reply:
Ah, yes this is to make the counter have a value of `2` to make it distinct from an off-by-one error from `0`. 😉 
Added a comment.